In [26]:
# Models Notebook
# Objective: Train/test split + baseline Logistic Regression


Load cleaned dataset

In [27]:
import pandas as pd

df = pd.read_csv("../data/titanic_clean.csv")
print(df.head())
print(df.dtypes)


   Survived  Pclass                                               Name  Sex  \
0         0       3                            Braund, Mr. Owen Harris    0   
1         1       1  Cumings, Mrs. John Bradley (Florence Briggs Th...    1   
2         1       3                             Heikkinen, Miss. Laina    1   
3         1       1       Futrelle, Mrs. Jacques Heath (Lily May Peel)    1   
4         0       3                           Allen, Mr. William Henry    0   

        Age  SibSp  Parch            Ticket      Fare  Embarked_Q  Embarked_S  \
0 -0.565736      1      0         A/5 21171 -0.502445       False        True   
1  0.663861      1      0          PC 17599  0.786845       False       False   
2 -0.258337      0      0  STON/O2. 3101282 -0.488854       False        True   
3  0.433312      1      0            113803  0.420730       False        True   
4  0.433312      0      0            373450 -0.486337       False        True   

   FamilySize  IsAlone  
0           2

Train/test split

In [28]:
from sklearn.model_selection import train_test_split

X = df.drop("Survived", axis=1)
y = df["Survived"]

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)


In [29]:
print(df.columns)
print(df.dtypes)


Index(['Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket',
       'Fare', 'Embarked_Q', 'Embarked_S', 'FamilySize', 'IsAlone'],
      dtype='object')
Survived        int64
Pclass          int64
Name           object
Sex             int64
Age           float64
SibSp           int64
Parch           int64
Ticket         object
Fare          float64
Embarked_Q       bool
Embarked_S       bool
FamilySize      int64
IsAlone         int64
dtype: object


In [30]:
df = df.drop(columns=["Name", "Ticket"], errors="ignore")
df["Embarked_Q"] = df["Embarked_Q"].astype(int)
df["Embarked_S"] = df["Embarked_S"].astype(int)
df["Age"] = df["Age"].fillna(df["Age"].median())
df["Fare"] = df["Fare"].fillna(df["Fare"].median())
print(df.dtypes)
print(df.isnull().sum())


Survived        int64
Pclass          int64
Sex             int64
Age           float64
SibSp           int64
Parch           int64
Fare          float64
Embarked_Q      int64
Embarked_S      int64
FamilySize      int64
IsAlone         int64
dtype: object
Survived      0
Pclass        0
Sex           0
Age           0
SibSp         0
Parch         0
Fare          0
Embarked_Q    0
Embarked_S    0
FamilySize    0
IsAlone       0
dtype: int64


Baseline Logistic Regression

In [31]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

X = df.drop("Survived", axis=1)
y = df["Survived"]

X_train, X_test, y_train, y_test = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

model = LogisticRegression(max_iter=1000)
model.fit(X_train, y_train)

y_pred = model.predict(X_test)

print("Accuracy:", accuracy_score(y_test, y_pred))
print("Confusion Matrix:\n", confusion_matrix(y_test, y_pred))
print("Classification Report:\n", classification_report(y_test, y_pred))


Accuracy: 0.8044692737430168
Confusion Matrix:
 [[97 13]
 [22 47]]
Classification Report:
               precision    recall  f1-score   support

           0       0.82      0.88      0.85       110
           1       0.78      0.68      0.73        69

    accuracy                           0.80       179
   macro avg       0.80      0.78      0.79       179
weighted avg       0.80      0.80      0.80       179



## Baseline Model Results
- Accuracy: ~0.78 (varies)
- Confusion matrix shows class balance
- Logistic regression is our baseline; future models will aim to beat this
